src/cmd/src/frontend.rs

```diff
@@ -149,18 +151,24 @@ impl TryFrom<StartCommand> for FrontendOptions {
 
         let tls_option = TlsOption::new(cmd.tls_mode, cmd.tls_cert_path, cmd.tls_key_path);
 
+        opts.http_options = Some(HttpOptions::default());
         if let Some(addr) = cmd.http_addr {
-            opts.http_options = Some(HttpOptions {
-                addr,
-                ..Default::default()
-            });
+            if let Some(ref mut http_options) = opts.http_options {
+                http_options.addr = addr;
+            }
         }
+
+        if let Some(ref mut http_options) = opts.http_options {
+            http_options.disable_dashboard = cmd.disable_dashboard
+        }
```

Comment:
```suggestion
        // init http options
        let mut http_options = HttpOptions {
            disable_dashboard: cmd.disable_dashboard,
            ..Default::default()
        };
        if let Some(addr) = cmd.http_addr {
            http_options.addr = addr;
        }
        opts.http_options = Some(http_options);
```

How about simplify initialization a little bit? :)